src/socket.rs

```diff
@@ -285,6 +285,51 @@ impl Socket {
         })
         .await
     }
+    pub async fn send(&self, buffer: &[u8]) -> Result<usize, Error> {
+        SocketFuture::new(|| {
+            #[cfg(feature = "defmt")]
+            defmt::trace!("Sending with socket {}", self.fd);
+
+            let mut send_result = unsafe {
+                nrfxlib_sys::nrf_send(self.fd, buffer.as_ptr() as *mut _, buffer.len() as u32, 0)
+            };
+
+            if send_result == -1 {
+                send_result = get_last_error().abs().neg();
+            }
+
+            #[cfg(feature = "defmt")]
+            defmt::trace!("Sending result {}", send_result);
+
+            const NRF_EWOULDBLOCK: i32 = -(nrfxlib_sys::NRF_EWOULDBLOCK as i32);
+
+            match send_result {
+                bytes_received @ 0.. => Poll::Ready(Ok(bytes_received as usize)),
+                NRF_EWOULDBLOCK => Poll::Pending,
+                error => Poll::Ready(Err(Error::NrfError(error))),
+            }
+        })
+        .await
+    }
+    pub fn set_option<'a>(&'a self, option: SocketOption<'a>) -> Result<(), Error> {
+        let length = option.get_length();
+
+        let result = unsafe {
+            nrfxlib_sys::nrf_setsockopt(
+                self.fd,
+                nrfxlib_sys::NRF_SOL_SECURE.try_into().unwrap(),
+                option.get_name(),
+                option.get_value(),
+                length as u32,
+            )
+        };
+
+        if result < 0 {
+            Err(Error::AddressNotFound)
```

Comment:
This error leaves a bit to be desired. There's much more info we can get out.
Here's the description: https://pubs.opengroup.org/onlinepubs/9699919799/functions/setsockopt.html 

For example, it would probably be good to make a special Error variant for `EISCONN` and `ENOPROTOOPT` and return the rest as an Nrfxlib error variant.

Reply:
whoops, that's a leftover from when I was hacking around.

I'll have a look